src/global/logger.ts

```diff
@@ -0,0 +1,171 @@
+    let formattedMessage = `[${timestamp}] ${level}: ${message}`;
+
+    if (data) {
+      try {
+        formattedMessage += `\n${JSON.stringify(data, null, 2)}`;
+      } catch (error) {
+        formattedMessage += `\n[JSON.stringify Error/Circular]`;
+      }
+    }
+
+    return formattedMessage;
+  }
+
+  private log(level: LogLevel, message: string, data?: any): void {
+    if (!this.shouldLog(LogLevelIndex[level])) {
+      return;
+    }
+
+    const logMessage: LogMessage = {
+      level,
+      message,
+      timestamp: this.getTimestamp(),
+      data,
+    };
+
+    const formattedMessage = this.formatMessage(logMessage);
+
+    switch (level) {
+      case 'DEBUG':
+        console.debug(formattedMessage);
+        break;
+      case 'INFO':
+        console.info(formattedMessage);
+        break;
+      case 'WARN':
+        console.warn(formattedMessage);
+        break;
+      case 'ERROR':
+      case 'FATAL':
+        console.error(formattedMessage);
+        break;
+      case 'OFF':
+        // Do nothing when logging is off
+        break;
+      default:
+        console.log(formattedMessage); // Default fallback for unknown log levels
+        break;
+    }
+  }
+
+  /**
+   * debug will be displayed when LogLevel level is set to DEBUG(5)
+   */
+  public debug(message: string, data?: any): void {
+    this.log('DEBUG', message, data);
+  }
+
+  /**
+   * info will be displayed when LogLevel level is set to DEBUG(5), INFO(4)
+   */
+  public info(message: string, data?: any): void {
+    this.log('INFO', message, data);
+  }
+
+  /**
+   * warn will be displayed when LogLevel level is set to DEBUG(5), INFO(4), WARN(3)
+   */
+  public warn(message: string, data?: any): void {
+    this.log('WARN', message, data);
+  }
+
+  /**
+   * error will be displayed when LogLevel level is set to DEBUG(5), INFO(4), WARN(3), ERROR(2)
+   */
+  public error(message: string, data?: any): void {
+    this.log('ERROR', message, data);
+  }
+
+  /**
+   * fatal will be displayed when LogLevel level is set to DEBUG(5), INFO(4), WARN(3), ERROR(2), FATAL(1)
+   */
+  public fatal(message: string, data?: any): void {
+    this.log('FATAL', message, data);
+  }
+
+  /**
+   * Set the logging level you would like system to display
+   * * 5 DEBUG  - show all logs
+   * * 4 INFO
+   * * 3 WARN
+   * * 2 ERROR
+   * * 1 FATAL
+   * * 0 OFF    - disable logs
+   */
+  public setLogLevel(level: LogLevel): void {
+    this.config.set('logLevel', level);
+  }
+
+  public getLogLevel(): LogLevel {
+    return this.config.get<string>('logLevel', 'INFO');
+  }
+
+  /**
+   *
+   * @returns logs levels displayed on the configured LogLevel
+   */
+  public getEnabledLogLevels() {
+    const logLevelStringKeys = Object.keys(LogLevelIndex).filter(
+      (v) => Number.isNaN(Number(v)) && v !== 'OFF'
+    );
+    const logLevels = logLevelStringKeys.filter((s) => {
+      return this.shouldLog(LogLevelIndex[s as LogLevel]);
+    });
```

Comment:
The `Number.isNaN(Number(v))` check doesn't seem necessary so the two filters could be merged.

Reply:
Leftover after refactoring from enum to const  pattern